### PR TITLE
new-app must validate --name instead of silently truncating and changing case

### PR DESCRIPTION
**Pull request by fabianofranz**
Fixes bugs 1247680 and 1251601.

Reverted `makeValidServiceName` to properly truncate and change case of generated service names and added cli validation for names provided by the user through `--name`.

### CODE REVIEW

**Comment by fabianofranz** [comment]
[test]

**Comment by fabianofranz** [comment]
@liggitt @smarterclayton please review.

File: pkg/generate/app/cmd/newapp.go [hunk 1]
```diff
@@ -426,6 +427,18 @@ func (c *AppConfig) detectSource(repositories []*app.SourceRepository) error {
...
 }
 
+var enforcedNameValidator = regexp.MustCompile("^[a-z]([-a-z0-9]*[a-z0-9])?$")
+
+const enforcedNameMinLength = 2
+const enforcedNameMaxLength = 24
+
+func (c *AppConfig) validateEnforcedName() error {
+	if len(c.Name) < enforcedNameMinLength || len(c.Name) > enforcedNameMaxLength || !enforcedNameValidator.MatchString(c.Name) {
+		return fmt.Errorf("invalid name: %s. Must be an a lower case alphanumeric (a-z, and 0-9) string with a maximum length of 24 characters, where the first character is a letter (a-z), and the '-' character is allowed anywhere except the first or last character.", c.Name)
```
Comment (smarterclayton) [inline pkg/generate/app/cmd/newapp.go]: Just call validateservicename here?

Comment (smarterclayton) [inline pkg/generate/app/cmd/newapp.go]: Or better, below

**Comment by fabianofranz** [comment]
@smarterclayton sure, fixed.

**Comment by fabianofranz** [comment]
[merge]

**Pull request merged**